pkg/storage/interfaces.go

```diff
@@ -1,187 +1,56 @@
-	f.Calls = append(f.Calls, FakeStatObjectCall{BucketName: bucketName, ObjectKey: objectKey})
-	return f.Fn(bucketName, objectKey)
+// Stat is the interface definition.
+func (f *FakeObjectStatter) Stat(ctx context.Context, path string) (storagedriver.FileInfo, error) {
+	f.Calls = append(f.Calls, FakeStatObjectCall{Path: path})
+	return f.Fn(ctx, path)
 }
 
-// ObjectPutter is a *(github.com/minio/minio-go).Client compatible interface, restricted to just
-// the PutObject function. You can use it in your code for easier unit testing without any
-// external dependencies.
-type ObjectPutter interface {
-	PutObject(bucketName, objectKey string, reader io.Reader, contentType string) (int64, error)
-}
-
-// FakePutObjectCall represents a single call to PutObject on a FakeObjectPutter.
-type FakePutObjectCall struct {
-	BucketName  string
-	ObjectKey   string
-	Reader      io.Reader
-	ContentType string
-}
-
-// FakeObjectPutter is a mock function that can be swapped in for an ObjectPutter, so you can
-// unit test your code.
-type FakeObjectPutter struct {
-	Fn    func(bucketName, objectKey string, reader io.Reader, contentType string) (int64, error)
-	Calls []FakePutObjectCall
-}
-
-// PutObject is the interface definition.
-func (f *FakeObjectPutter) PutObject(bucketName, objectKey string, reader io.Reader, contentType string) (int64, error) {
-	f.Calls = append(f.Calls, FakePutObjectCall{
-		BucketName:  bucketName,
-		ObjectKey:   objectKey,
-		Reader:      reader,
-		ContentType: contentType,
-	})
-	return f.Fn(bucketName, objectKey, reader, contentType)
-}
-
-// Object is a *(github.com/minio/minio-go).Object compatible interface. Currently, ObjectGetter
-// returns these so that FakeObjectGetters can return mock implementations.
-type Object interface {
-	// This is called an interface composition - it automatically gives your interface the function
-	// in io.Reader (https://godoc.org/io#Reader) and the function in io.Closer
-	// (https://godoc.org/io#Closer).
-	io.ReadCloser
-	// This is also an interface composition. It gives your interface the function in
-	// io.Seeker (https://godoc.org/io#Seeker).
-	io.Seeker
-	// This is also interface composition. It gives your interface the function in
-	// io.ReaderAt (https://godoc.org/io#ReaderAt).
-	io.ReaderAt
-	// This function is the last one we have to add to make this interface have all the same
-	// functions as s3.Object.
-	Stat() (s3.ObjectInfo, error)
-}
-
-// ObjectGetter is the interface to get an object from object storage. The minio client doesn't
-// already satisfy this interface, because the GetObject func
-// (https://godoc.org/github.com/minio/minio-go#Client.GetObject) doesn't return an Object.
-// Instead, it returns a *s3.Object. Use the RealObjectGetter below to use the minio client.
+// ObjectGetter is a *(github.com/docker/distribution/registry/storage/driver/factory).StorageDriverFactory compatible interface, restricted to
```

Comment:
Isn't this a `*(github.com/docker/distribution/registry/storage/driver).StorageDriver` compatible interface?